src/composite/tag/tag.jl

```diff
@@ -8,11 +8,14 @@ but tag the block with some extra information.
 """
 abstract type TagBlock{BT, N, T} <: AbstractContainer{BT, N, T} end
 
+# forward content properties
 cache_key(tb::TagBlock) = cache_key(content(tb))
 occupied_locs(x::TagBlock) = occupied_locs(content(x))
 
 Base.:(==)(a::TB, b::TB) where {TB<:TagBlock} = content(a) == content(b)
+Base.length(x::TagBlock) = length(content(x))
 Base.getindex(c::TagBlock, index...) = getindex(content(c), index...)
 Base.setindex!(c::TagBlock, val, index...) = setindex!(content(c), val, index...)
 Base.iterate(c::TagBlock) = iterate(content(c))
 Base.iterate(c::TagBlock, st) = iterate(content(c), st)
+Base.eltype(x::TagBlock) = eltype(content(x))
```

Comment:
Why not using the default `eltype`?

Reply:
`eltype` should be forwarded for iteration protocal

Reply:
ok, I see. But why iterating over tag block is nessesary? see the example below.